samples/bluetooth/peripheral_uart/src/main.c

```diff
@@ -227,11 +258,41 @@ static int uart_init(void)
 
 	k_work_init_delayable(&uart_work, uart_work_handler);
 
+
+	if (IS_ENABLED(CONFIG_UART_INTERRUPT_DRIVEN) && !uart_test_async_api(uart)) {
+		/* Implement API adapter */
+		uart_async_adapter_init(async_adapter, uart);
+		uart = async_adapter;
+	}
 	err = uart_callback_set(uart, uart_cb, NULL);
 	if (err) {
+		LOG_ERR("Cannot initialize UART callback");
 		return err;
 	}
 
+	if (IS_ENABLED(CONFIG_UART_LINE_CTRL)) {
+		LOG_INF("Wait for DTR");
+		while (true) {
```

Comment:
Well.... the line state is not returned as a return value - it is written to pointed variable.
We can use comma operator and write:

```
while(!(uart_line_ctrl_get(uart, UART_LINE_CTRL_DTR, &dtr), dtr)) {
}
```

But I am sure it would be considered to be odd and hard to read. Personally I like short forms, but the one used now is more obvious for beginners and less error prone.

Reply:
go it